Lib/shutil.py

```diff
@@ -170,6 +171,35 @@ def _copyxattr(src, dst, *, follow_symlinks=True):
     def _copyxattr(*args, **kwargs):
         pass
 
+if sys.platform == 'darwin':
+    # As of macOS 10.10, the `copyfile` API changed to not copy certain flags
+    mac_ver = tuple(int(x) for x in platform.mac_ver()[0].split('.'))
+    if mac_ver >= (10, 10, 0):
```

Comment:
I'd compare against os.uname().release instead of using platform.mac_ver as this is related to a kernel API. The uname.release to compare against is 14.0.

Better yet: guard against the availability of os.SF_RESTRICTED instead.